CRM/Logging/Schema.php

```diff
@@ -776,8 +780,16 @@ public function triggerInfo(&$info, $tableName = NULL, $force = FALSE) {
         $sqlStmt .= "NEW.$column, ";
         $deleteSQL .= "OLD.$column, ";
       }
-      $sqlStmt .= "CONNECTION_ID(), @civicrm_user_id, '{eventName}');";
-      $deleteSQL .= "CONNECTION_ID(), @civicrm_user_id, '{eventName}');";
+      if (civicrm_api3('Setting', 'getvalue', array('name' => 'logging_uniqueid_date', 'group' => 'core'))) {
+        $sqlStmt .= "@uniqueID, @civicrm_user_id, '{eventName}');";
+        $deleteSQL .= "@uniqueID, @civicrm_user_id, '{eventName}');";
+      }
+      else {
+        // The log tables have not yet been converted to have varchar(17) fields for log_conn_id.
+        // Continue to use the less reliable connection_id for al tables for now.
+        $sqlStmt .= "CONNECTION_ID(), @civicrm_user_id, '{eventName}');";
+        $deleteSQL .= "@uniqueID, @civicrm_user_id, '{eventName}');";
```

Comment:
That does indeed look like an error! will push a fix. Not clear why I haven't hit is - perhaps my 4.6 version does not have that